lib/contentful-util.js

```diff
@@ -27,12 +27,14 @@ function normalizeEntry({ contentTypes, entry, defaultLocaleCode, options }) {
         if (!entry.fields.file) {
             return null;
         }
-        const file = getLocalizedFieldValue(entry.fields.file, defaultLocaleCode);
+        let file = getLocalizedFieldValue(entry.fields.file, defaultLocaleCode);
+        if (typeof file !== 'object') file = entry.fields.file;
 
         fields = {
             contentType: file.contentType,
             fileName: file.fileName,
-            url: `https:${file.url}`
+            url: `https:${file.url}`,
+            details: file.details
```

Comment:
main changes here
`        if (typeof file !== 'object') file = entry.fields.file;`
is needed because for a file object like this (obtained from api):
```js
            file: {
                url: '//images.ctfassets.net/spaceId/7orLdboQQowIUs22KAW4U/e237faf0c6a0c89f8dce3e35e552176e/matt-palmer-254999.jpg',
                details: {
                    size: 2293094,
                    image: {
                        width: 3000,
                        height: 2000
                    }
                },
                fileName: 'matt-palmer-254999.jpg',
                contentType: 'image/jpeg'
            }
```
it was returning the url value (which is just one of the fields in the file) which would break the return object

I'm not completely aware of all the localization nuances for contentful but it didnt look right to me

Reply:
Hi @UltimateForm 
Thank you for support and collaboration!

re: `if (typeof file !== 'object') file = entry.fields.file`
Can you provide steps to reproduce the case where Contentful return `file` field not as object?

According to Contentful API, all fields returned from /sync API always localized:
> Syncing entries or assets returns all available localizations instead of a single one. Usually resources returned from the delivery API have only a single value per field, the value of the locale you requested or the default locale, **but the sync endpoint returns all locales per field**.
docs: https://www.contentful.com/developers/docs/concepts/sync/#localization

So if the `entry.fields.file` exists, it will always be a localized object:
```js
{
  file: {
    "en-US": { url: "...", details: "...", ...},
    "fr-FR": { url: "...", details: "...", ...}
  }
}
```
<img width="488" alt="image" src="https://user-images.githubusercontent.com/97896/167376302-4899a7fe-0679-4f9a-993b-2fc856257525.png">

Then the `getLocalizedFieldValue` returns the object for the `defaultLocale`, or the first locale it finds if no default locale provided.

Reply:
re: `details: file.details`

I suggest flattening `details` into generic field names to be consistent with other sourcebit plugins:
```
{
  contentType: file.contentType,
  fileName: file.fileName,
  url: `https:${file.url}`,
  size: file.details.size,
  dimensions: file.details.image
}
```

Reply:
done

> re: `details: file.details`
> 
> I suggest flattening `details` into generic field names to be consistent with other sourcebit plugins:
> 
> ```js
> {
>   contentType: file.contentType,
>   fileName: file.fileName,
>   url: `https:${file.url}`,
>   size: file.details.size,
>   dimensions: file.details.image
> }
> ```

sounds good to me, changed tests to reflect this change

```js
        expect(objects[3].size).toBe(MOCK_ASSETS[0].fields.file.details.size);
        expect(objects[3].dimensions).toBeDefined();
        expect(objects[3].dimensions.width).toBe(MOCK_ASSETS[0].fields.file.details.image.width);
        expect(objects[3].dimensions.height).toBe(MOCK_ASSETS[0].fields.file.details.image.height);
```

Reply:
@smnh as for the file object turns out i looking at the wrong api 😀 you are correct, the sync api always returns the localized structure, i've made changes to correct the tests and revert the contentful-util.js changes